src/garage/tf/embeddings/base.py

```diff
@@ -0,0 +1,275 @@
+"""Base class for embedding networks in TensorFlow."""
+import abc
+
+import tensorflow as tf
+
+from garage.misc.tensor_utils import flatten_tensors, unflatten_tensors
+
+
+class EmbeddingSpec:
+    """Specification of an embedding network.
+
+    Args:
+        input_space (akro.Space): The input space of the embedding.
+        latent_space (akro.Space): The latent space of the embedding.
+
+    """
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, input_space, latent_space):
+        self.input_space = input_space
+        self.latent_space = latent_space
+
+
+class Embedding(abc.ABC):
```

Comment:
i think that this should probably be called Encoder

A Decoder maps from a low-dimensional space to a high-dim one.

An embedding might be implemented using an encoder-decoder pair, just an encoder, or just a decoder.